src/main/java/com/aws/greengrass/telemetry/TelemetryAgent.java

```diff
@@ -246,19 +246,25 @@ void aggregatePeriodicMetrics() {
     /**
      * Helper for metrics uploader. Also used in tests.
      */
+    @SuppressWarnings("PMD.AvoidCatchingThrowable")
     void publishPeriodicMetrics() {
-        if (!isConnected.get()) {
-            logger.atDebug().log("Cannot publish the metrics. MQTT connection interrupted.");
-            return;
+        try {
+            if (!isConnected.get()) {
+                logger.atDebug().log("Cannot publish the metrics. MQTT connection interrupted.");
```

Comment:
nit - you don't need the trailing "." - it will print "MQTT connection interrupted.."